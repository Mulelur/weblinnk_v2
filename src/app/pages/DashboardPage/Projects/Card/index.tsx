import React from 'react';
import styled from 'styled-components/macro';
import { CardActions } from './CardActions';
import { CardContent } from './CardContent';
import { CardHeader } from './CardHeader';
import { CardMedia } from './CardMedia';

type Props = {
  imageSrc: string;
};

export function Card(props: Props) {
  const { imageSrc } = props;
  return (
    <Wrapper>
      <CardHeader />
      <CardMedia src={imageSrc} />
      <CardContent />
      <CardActions />
    </Wrapper>
  );
}

const Wrapper = styled.div`
  width: 28rem;
  border-radius: 0.75rem;
  background-color: ${p => p.theme.background};
  border: 1px solid ${p => p.theme.borderLight};
  box-shadow: 0px 4px 6px -1px rgba(0, 0, 0, 0.1),
    0px 2px 4px -1px rgba(0, 0, 0, 0.06);
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-direction: column;
`;
