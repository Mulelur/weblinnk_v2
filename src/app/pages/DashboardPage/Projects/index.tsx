import * as React from 'react';
import styled from 'styled-components/macro';
import { Title } from '../components/Title';
// import { SubTitle } from '../components/SubTitle';
// import { P } from '../components/P';
// import { A } from 'app/components/A';
// import { GithubRepoForm } from './GithubRepoForm';
// import { LanguageSwitch } from './LanguageSwitch';
import { ReactComponent as AddIcon } from './assets/add-icon.svg';

// import { useTranslation } from 'react-i18next';
// import { Link } from 'app/components/Link';
// import { messages } from '../messages';
import { Card } from './Card';
import { Button } from './components/Button';

export function Projects() {
  return (
    <>
      <Title as="h2">Projects</Title>
      <List>
        <Project>
          <Card imageSrc="https://images.unsplash.com/photo-1651724682401-a10daf761c96?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1771&q=80" />
        </Project>
        <Project>
          <Card imageSrc="https://images.unsplash.com/photo-1651648814980-55936f67f9f7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80" />
        </Project>
        <Project>
          <Card imageSrc="https://images.unsplash.com/photo-1651860207580-ddc55836bd8a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80" />
        </Project>
        <Project>
          <Card imageSrc="https://images.unsplash.com/photo-1636837955417-2d8a4e49368f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80" />
        </Project>
      </List>
      <Button>
        <AddIcon />
      </Button>
    </>
  );
}

const List = styled.div`
  width: 100%;
  margin: 2.15rem 0 6rem 0;
  display: inline-grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fit, minmax(21rem, 1fr));
  padding-bottom: 4rem;
`;

const Project = styled.div`
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: flex;
  justify-content: center;
  align-items: center;
`;
