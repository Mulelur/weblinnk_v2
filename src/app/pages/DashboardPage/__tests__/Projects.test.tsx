import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Projects } from '../Projects';
import i18next from 'i18next';

const shallowRenderer = createRenderer();

describe('<Projects />', () => {
  it.only('should render with en translations', () => {
    i18next.changeLanguage('en');
    shallowRenderer.render(<Projects />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render with de translations', () => {
    i18next.changeLanguage('de');
    shallowRenderer.render(<Projects />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render different after language change', () => {
    i18next.changeLanguage('en');
    shallowRenderer.render(<Projects />);
    const renderedOutput1 = shallowRenderer.getRenderOutput();

    i18next.changeLanguage('de');
    shallowRenderer.render(<Projects />);
    const renderedOutput2 = shallowRenderer.getRenderOutput();

    expect(renderedOutput1).not.toEqual(renderedOutput2);
  });
});
