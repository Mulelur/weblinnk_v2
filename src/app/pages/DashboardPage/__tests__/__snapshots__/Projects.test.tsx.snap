// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render with en translations 1`] = `
<React.Fragment>
  <Title
    as="h2"
  >
    Projects
  </Title>
  <Projects__List>
    <Projects__Project>
      <Card
        imageSrc="https://images.unsplash.com/photo-1651724682401-a10daf761c96?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1771&q=80"
      />
    </Projects__Project>
    <Projects__Project>
      <Card
        imageSrc="https://images.unsplash.com/photo-1651648814980-55936f67f9f7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80"
      />
    </Projects__Project>
    <Projects__Project>
      <Card
        imageSrc="https://images.unsplash.com/photo-1651860207580-ddc55836bd8a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80"
      />
    </Projects__Project>
    <Projects__Project>
      <Card
        imageSrc="https://images.unsplash.com/photo-1636837955417-2d8a4e49368f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80"
      />
    </Projects__Project>
  </Projects__List>
  <Button>
    <ForwardRef(SvgAddIcon) />
  </Button>
</React.Fragment>
`;
